### PR TITLE
[Core] Add service_name.json

### PR DESCRIPTION
**Pull request by qwordy**
Add a file `service_name.json`. It maintains service information of commands.

Twin PR, https://github.com/Azure/azure-cli/pull/17159. Visit it for more details.

Example
```
  {
    "Command": "az databox",
    "Description": "Microsoft Azure Command-Line Tools DataBox Extension",
    "AzureServiceName": "Azure Databox",
    "ConversationalServiceName": "Databox",
    "URL": "https://docs.microsoft.com/azure/databox/"
  },
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/service_name.json [hunk 1]
```diff
@@ -0,0 +1,422 @@
+[
+  {
+    "Command": "az ai-did-you-mean-this",
```
Comment (yungezz) [inline src/service_name.json:3]: I believe this extension is deprecated already, could you pls double confirm?

File: src/service_name.json [hunk 2]
```diff
@@ -0,0 +1,422 @@
...
+    "Description": "Recommend recovery options on failure.",
+    "AzureServiceName": "Microsoft AI",
+    "ConversationalServiceName": "Microsoft AI",
+    "URL": ""
+  },
+  {
+    "Command": "az ai-examples",
```
Comment (yungezz) [inline src/service_name.json:10]: how to maintain this file? what's process when new extension is released?

Comment (qwordy) [reply src/service_name.json:10]: Manual. We can add a linter rule to force it.
I added a new column "ExtensionName". It is required when "AzureCLIReferenceType" is Extension or "hasExtension" is 1. This column is used to check if all extensions have records.

File: src/service_name.json [hunk 3]
```diff
@@ -0,0 +1,514 @@
...
+    "ExtensionName": "ai-examples",
+    "URL": "https://docs.microsoft.com/archive/msdn-magazine/2017/connect/artificial-intelligence-getting-started-with-microsoft-ai"
+  },
+  {
+    "Command": "az alias",
+    "Description": "Support for command aliases",
+    "AzureServiceName": "Azure CLI",
+    "ConversationalServiceName": "Azure CLI",
+    "ExtensionName": "alias",
+    "URL": "https://docs.microsoft.com/cli/azure/"
```
Comment (fengzhou-msft) [inline src/service_name.json:16]: Where does this URL come from? The current one is a general CLI doc link.

Comment (qwordy) [reply src/service_name.json:16]: We can set it reference of `az alias`

File: src/service_name.json [hunk 4]
```diff
@@ -0,0 +1,514 @@
...
+    "URL": "https://azure.microsoft.com/support/options/"
+  },
+  {
+    "Command": "az timeseriesinsights",
+    "Description": "Microsoft Azure Command-Line Tools TimeSeriesInsightsClient Extension",
+    "AzureServiceName": "Azure Time Series Insights",
+    "ConversationalServiceName": "Time Series Insights",
+    "ExtensionName": "timeseriesinsights",
+    "URL": "https://docs.microsoft.com/azure/time-series-insights/"
+  },
```
Comment (kairu-ms) [inline src/service_name.json]: This command group is deprecated. Now its name is "az tsi" which already included below.
```suggestion
```

File: src/service_name.json [hunk 5]
```diff
@@ -0,0 +1,514 @@
...
+    "URL": "https://docs.microsoft.com/azure/aks/"
+  },
+  {
+    "Command": "az k8sconfiguration",
+    "Description": "Microsoft Azure Command-Line Tools K8sconfiguration Extension",
+    "AzureServiceName": "Azure Kubernetes Service",
+    "ConversationalServiceName": "Kubernetes Service",
+    "ExtensionName": "k8sconfiguration",
+    "URL": "https://docs.microsoft.com/azure/aks/"
+  },
```
Comment (kairu-ms) [inline src/service_name.json]: I think `k8s-configuration` and `k8sconfiguration` are the same.

**Review by dbradish-microsoft — DISMISSED** [review]

File: src/service_name.json [hunk 6]
```diff
@@ -0,0 +1,450 @@
...
+    "Command": "az communication",
+    "Description": "Manage communication services",
+    "AzureServiceName": "Azure Communication Service",
+    "ExtensionName": "communication",
+    "URL": "https://docs.microsoft.com/azure/communication-services/"
+  },
+  {
+    "Command": "az confluent",
+    "Description": "Manage confluent resources.",
+    "AzureServiceName": "",
```
[the remark this answers is not in the document]
Comment (qwordy) [reply src/service_name.json:75]: Have not found an appropriate value

File: src/service_name.json [hunk 7]
```diff
@@ -0,0 +1,450 @@
...
+    "ExtensionName": "support",
+    "URL": "https://azure.microsoft.com/support/options/"
+  },
+  {
+    "Command": "az timeseriesinsights",
+    "Description": "Microsoft Azure Command-Line Tools TimeSeriesInsightsClient Extension",
+    "AzureServiceName": "Azure Time Series Insights",
+    "ExtensionName": "timeseriesinsights",
+    "URL": "https://docs.microsoft.com/azure/time-series-insights/"
+  },
```
Comment (qwordy) [inline src/service_name.json:435]:
```suggestion
```

File: src/service_name.json [hunk 8]
```diff
@@ -0,0 +1,450 @@
...
+  {
+    "Command": "az ai-examples",
+    "Description": "Add AI powered examples to help content.",
+    "AzureServiceName": "Microsoft AI",
+    "ExtensionName": "ai-examples",
+    "URL": "https://docs.microsoft.com/archive/msdn-magazine/2017/connect/artificial-intelligence-getting-started-with-microsoft-ai"
+  },
+  {
+    "Command": "az alias",
+    "Description": "Support for command aliases",
```
[the remark this answers is not in the document]
Comment (qwordy) [reply src/service_name.json:11]: Replied in https://github.com/Azure/azure-cli/pull/17159

File: src/service_name.json (hunk 5 above)
Comment (qwordy) [reply src/service_name.json]: Unfortunately, they are two extension. Ugly naming.
```
k8s-configuration               1.0.0      Microsoft Azure Command-Line Tools K8s-configuration Extension                                                     False      False           False
k8sconfiguration                0.2.4      Microsoft Azure Command-Line Tools K8sconfiguration Extension                                                      True       False           False
```

File: src/service_name.json [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+    "AzureServiceName": "Azure CLI",
+    "ExtensionName": "alias",
+    "URL": "https://docs.microsoft.com/cli/azure/"
```
Comment (qwordy) [inline src/service_name.json:14]:
```suggestion
    "URL": ""
```

**Comment by qwordy** [comment]
@dbradish-microsoft Can you revoke the "request change"? Thanks.

**Comment by dbradish-microsoft** [comment]
@qwordy , done.

**Comment by qwordy** [comment]
I moved content to these 2 PRs.
https://github.com/Azure/azure-cli/pull/17428
https://github.com/Azure/azure-cli-extensions/pull/3186

**Comment by qwordy** [comment]
Reopen this PR. The following two new PRs that contain linter and service_name.json are still in review. Reviewing Azure CLI Core code is always hard. As a result, I decide to split linter and content into separate PRs. Let's merge this PR first.
https://github.com/Azure/azure-cli/pull/17428
https://github.com/Azure/azure-cli-extensions/pull/3186

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**